### PR TITLE
Infernce Model string support of TFNet

### PR DESCRIPTION
**Pull request by Litchilitchy**
One feature lost for this PR, inferenceModel could not print batchSize anymore, user should print them by themselves before input the Tensor to model.

### CODE REVIEW

**Comment by jason-dai** [comment]
Please provide some descriptions/unit tests on how the user is supposed to use this features? E.g., what the user input looks like when sending the requests to cluster serving.

File: zoo/src/main/scala/com/intel/analytics/zoo/pipeline/inference/InferenceModel.scala [hunk 1]
```diff
@@ -743,17 +744,17 @@ class InferenceModel(private var autoScalingEnabled: Boolean = true,
     val model: AbstractModel = retrieveModel()
     try {
       val begin = System.nanoTime()
-      val batchSize = if (inputActivity.isTensor) {
-        inputActivity.toTensor[Float].size(1)
-      } else {
-        val sampleKey = inputActivity.toTable.keySet.head
-        inputActivity.toTable(sampleKey).asInstanceOf[Tensor[Float]].size(1)
-      }
+//      val batchSize = if (inputActivity.isTensor) {
```
Comment (jason-dai) [inline zoo/src/main/scala/com/intel/analytics/zoo/pipeline/inference/InferenceModel.scala:747]: remove unused code

File: zoo/src/test/scala/com/intel/analytics/zoo/serving/InferenceSpec.scala [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+ * limitations under the License.
+ */
+
+package com.intel.analytics.zoo.serving
+
+import com.intel.analytics.bigdl.tensor.Tensor
+import com.intel.analytics.zoo.serving.utils.{ClusterServingHelper, SerParams}
+import org.scalatest.{FlatSpec, Matchers}
+
+class InferenceSpec extends FlatSpec with Matchers {
```
Comment (jason-dai) [inline zoo/src/test/scala/com/intel/analytics/zoo/serving/InferenceSpec.scala:23]: is this testing anything?

File: pyzoo/zoo/serving/client.py [hunk 3]
```diff
@@ -101,8 +101,16 @@ def enqueue(self, uri, **data):
         field_list = []
         data_list = []
         for key, value in data.items():
+            if "string" in key:
+                # list of string will be converted to Tensor of String
+                # use | to split
+                str_concat = '|'.join(value)
```
Comment (jason-dai) [inline pyzoo/zoo/serving/client.py:107]: what if there is '|' in the original strings?

File: zoo/src/test/scala/com/intel/analytics/zoo/serving/InferenceSpec.scala (hunk 2 above)
Comment (Litchilitchy) [reply zoo/src/test/scala/com/intel/analytics/zoo/serving/InferenceSpec.scala:23]: This is TODO, nothing is tested so far, this need some string schema currently unknown, will check.

File: pyzoo/zoo/serving/client.py (hunk 3 above)
Comment (Litchilitchy) [reply pyzoo/zoo/serving/client.py:107]: Will check schema, if so, will change to another one.

File: zoo/src/main/scala/com/intel/analytics/zoo/pipeline/inference/InferenceModel.scala (hunk 1 above)
Comment (Litchilitchy) [reply zoo/src/main/scala/com/intel/analytics/zoo/pipeline/inference/InferenceModel.scala:747]: Noted

**Comment by Litchilitchy** [comment]
> Please provide some descriptions/unit tests on how the user is supposed to use this features? E.g., what the user input looks like when sending the requests to cluster serving.

This will update to doc

**Pull request merged**